### PR TITLE
Make Lua a hard requirement for rpm

### PR DESCRIPTION
**Pull request by pmatilai**
More and more macros, scriptlets and other functionality has been getting
built around Lua, to the point that it has in practise been required for
several years now.

Maintaining the pretense of being optional comes at a cost in holding
back developments and having to check for that theoretical special
case. Lets make it a hard requirement and embrace it some more!

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

File: luaext/Makefile.am [hunk 1]
```diff
@@ -5,8 +5,8 @@ EXTRA_DIST = \
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir) -I$(top_builddir)/include -I.
 
-libluaext_la_CPPFLAGS = $(AM_CPPFLAGS) -DWITH_POSIX @LUA_CFLAGS@
-libluaext_la_LIBADD = @LUA_LIBS@
+libluaext_la_CPPFLAGS = $(AM_CPPFLAGS) -DWITH_POSIX $(LUA_CFLAGS)
+libluaext_la_LIBADD = $(LUA_LIBS)
 libluaext_la_SOURCES = \
```
Comment (mlschroe) [inline luaext/Makefile.am:10]: What's with this change? The rest of the code still uses the @LUA_XXX@ syntax, so why is this done here?
(ZSTD_XXX is the other odd candidate here)

Comment (pmatilai) [reply luaext/Makefile.am:10]: Oh, good catch. I initially looked at going the ZSTD-route, but then switched back but this one case was left behind.

Comment (pmatilai) [reply luaext/Makefile.am]: This change dropped in the last push.

**Review by Conan-Kudo — APPROVED** [review]

**Comment by Conan-Kudo** [comment]
@pmatilai small nit of the commit message: "practise" -> "practice" (en-gb has the former for verb, but you're not using it that way, and en-us doesn't have that spelling at all).

**Comment by pmatilai** [comment]
Right, I have to say I seem to have forgotten that particular nuance of the English language. Will fix, but I have to say I was expecting somewhat more fundamental discussion over this matter :laughing:

**Comment by Conan-Kudo** [comment]
Nah, this makes a ton of sense to do. At some point, I need to figure out how to do Lua for debbuild too. I just don't know how I'll square that with how Debian package scripts work...

**Comment by pmatilai** [comment]
Spelling fixed...

**Comment by pmatilai** [comment]
Okay so nobody objects, even remotely, to the actual change. Except for myself, for this will make bisecting old stuff somewhat more difficult, but then ability to disable Lua only helps if the thing to bisect is not Lua. There's a price to everything, shrug.

**Pull request merged**